example/metaball.js

```diff
@@ -142,66 +136,51 @@ const position = (time, i) => {
 }
 let start_bounds, end_bounds
 [start_bounds, end_bounds] = bounds
-let step_sizes = [0, 1, 2].map((i) => (end_bounds[i] - start_bounds[i]) / size)
+const step_sizes = [0, 1, 2].map((i) => (end_bounds[i] - start_bounds[i]) / size)
 const r = size * Math.sqrt(strength / subtract)
 
 const render = (tick) => {
   let time = 0.05 * tick
-  let fieldArray = ndarray(new Float32Array(size * size * size), [size, size, size])
-  let balls = Array(numblobs).fill().map((_, i) => {
-    let bx, by, bz
-    [bx, by, bz] = position(time, i)
-    return {
-      functional: metaball(bx, by, bz, strength, subtract),
-      center: [bx, by, bz]
-    }
-  })
-
-  fill(fieldArray, (i, j, k) => {
```

Comment:
@mikolalysenko for some reason when I prepare `fieldArray` with data before initializing an `ndarray` it is far more performant than using `fill`. I don't understand why, do you have insight on this? The implementation of `fill` is pretty bare-bones, so I don't see where the performance hit comes from. Is it perhaps related to the inherent cost to call a function in JS?